src/main/java/com/flagsmith/FlagsmithApiWrapper.java

```diff
@@ -1,176 +1,216 @@
+import java.util.concurrent.TimeoutException;
+import lombok.Data;
 import okhttp3.HttpUrl;
 import okhttp3.MediaType;
 import okhttp3.Request;
 import okhttp3.RequestBody;
-import okhttp3.Response;
 
-class FlagsmithApiWrapper implements FlagsmithSdk {
+@Data
+public class FlagsmithApiWrapper implements FlagsmithSdk {
 
   private static final String AUTH_HEADER = "X-Environment-Key";
   private static final String ACCEPT_HEADER = "Accept";
+  private static final Integer TIMEOUT = 15000;
+  private static final String FLAG_KEY = "feature-flag";
+
   private final FlagsmithLogger logger;
   private final FlagsmithConfig defaultConfig;
   private final HashMap<String, String> customHeaders;
   // an api key per environment
   private final String apiKey;
+  private final RequestProcessor requestor;
+  private FlagsmithCache cache = null;
+
+  /**
+   * Instantiate with cache.
+   *
+   * @param cache cache object
+   * @param defaultConfig config object
+   * @param customHeaders custom headers list
+   * @param logger logger object
+   * @param apiKey api key
+   */
+  public FlagsmithApiWrapper(
+      final FlagsmithCache cache,
+      final FlagsmithConfig defaultConfig,
+      final HashMap<String, String> customHeaders,
+      final FlagsmithLogger logger,
+      final String apiKey
+  ) {
+    this(defaultConfig, customHeaders, logger, apiKey);
+    this.cache = cache;
+  }
 
-  public FlagsmithApiWrapper(final FlagsmithConfig defaultConfig,
+  /**
+   * Instantiate with config, custom headers, logger and apikey.
+   * @param defaultConfig config object
+   * @param customHeaders custom headers list
+   * @param logger logger instance
+   * @param apiKey api key
+   */
+  public FlagsmithApiWrapper(
+      final FlagsmithConfig defaultConfig,
       final HashMap<String, String> customHeaders,
       final FlagsmithLogger logger,
-      final String apiKey) {
+      final String apiKey
+  ) {
     this.defaultConfig = defaultConfig;
     this.customHeaders = customHeaders;
     this.logger = logger;
     this.apiKey = apiKey;
+    requestor = new RequestProcessor(
+        defaultConfig.getHttpClient(),
+        logger,
+        defaultConfig.getRetries()
+    );
   }
 
-  @Override
-  public FlagsAndTraits getFeatureFlags(FeatureUser user, boolean doThrow) {
-    HttpUrl.Builder urlBuilder;
-    if (user == null) {
-      urlBuilder = defaultConfig.flagsUri.newBuilder()
-          .addEncodedQueryParameter("page", "1");
-    } else {
-      return getUserFlagsAndTraits(user, doThrow);
-    }
+  /**
+   * Get Feature Flags from API.
+   *
+   * @param doThrow - whether throw exception or not
+   * @return
+   */
+  public List<FeatureStateModel> getFeatureFlags(boolean doThrow) {
+    List<FeatureStateModel> featureFlags = null;
 
-    final Request request = this.newRequestBuilder()
-        .url(urlBuilder.build())
-        .build();
+    if (cache != null) {
+      featureFlags = cache.getIfPresent(FLAG_KEY);
```

Comment:
What does the `FLAG_KEY` represent - it feels like it is related to a single flag rather than the entire environment flags list?

Reply:
This has been changed to use CacheEnvironmentFlagKey() already present in the Cache interface.